code/modules/projectiles/guns/projectile.dm

```diff
@@ -176,8 +191,13 @@
 	load_ammo(A, user)
 
 /obj/item/weapon/gun/projectile/attack_self(mob/user as mob)
-	if(firemodes.len > 1)
-		switch_firemodes(user)
+	if(is_jammed)
+		user.visible_message("\The [user] unjams \the [src]!")

```

Comment:
I'd say, wrap the text into a `<span class='notice'></span>` here. Black text on the logs is really odd.